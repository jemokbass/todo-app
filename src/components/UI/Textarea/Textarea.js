import React, { forwardRef } from 'react';
import PropTypes from 'prop-types';

const Textarea = forwardRef((props, ref) => {
  const {
    autoCorrect,
    autoComplete,
    onBlur,
    onChange,
    label,
    disabled,
    name,
    className,
    errors,
    errorsMessage,
    placeholder,
  } = props;
  const textareaClassName = className ? ` ${className}` : '';

  return (
    <label className="label textarea-field">
      <span className="textarea-field__label">{label}</span>
      <textarea
        className={`textarea${textareaClassName}`}
        autoComplete={autoComplete}
        autoCorrect={autoCorrect}
        ref={ref}
        onChange={onChange}
        onBlur={onBlur}
        disabled={disabled}
        name={name}
        placeholder={placeholder}
      />
      {errors && <span className="input-field__error">{errorsMessage}</span>}
    </label>
  );
});

Textarea.defaultProps = {
  label: '',
  onChange: () => {},
  onBlur: () => {},
  autoComplete: 'on',
  autoCorrect: 'on',
  disabled: false,
  name: '',
  placeholder: '',
};

Textarea.propTypes = {
  label: PropTypes.string,
  onChange: PropTypes.func,
  onBlur: PropTypes.func,
  autoComplete: PropTypes.oneOf(['off', 'on']),
  autoCorrect: PropTypes.oneOf(['off', 'on']),
  disabled: PropTypes.bool,
  name: PropTypes.string,
  value: PropTypes.string,
  placeholder: PropTypes.string,
};

export default Textarea;
