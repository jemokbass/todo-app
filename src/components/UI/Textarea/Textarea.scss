.textarea-field {
  display: block;
  margin: 0 0 10px;

  &__label {
    display: block;
    font-size: 22px;
  }

  .textarea {
    display: block;
    width: 100%;
    min-height: 120px;
    resize: none;
  }

  .textarea[disabled] {
    opacity: 0.6;
  }
}
