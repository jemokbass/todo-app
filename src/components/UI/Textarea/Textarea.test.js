import React from 'react';
import { render, screen } from '@testing-library/react';

import Textarea from './Textarea';

describe('Textarea UI component', () => {
  it('Should render textarea component', () => {
    render(<Textarea />);
  });

  it('Should render disabled textarea component', () => {
    render(<Textarea disabled placeholder="textarea" />);
    expect(screen.getByPlaceholderText('textarea')).toBeDisabled();
  });

  describe('Should render snapshots', () => {
    it('Should render empty textarea snapshot', () => {
      const { asFragment } = render(<Textarea />);
      expect(asFragment(<Textarea />)).toMatchSnapshot();
    });

    it('Should render disabled textarea snapshot', () => {
      const { asFragment } = render(<Textarea disabled />);
      expect(asFragment(<Textarea disabled />)).toMatchSnapshot();
    });
  });

  describe('Default props', () => {
    it('Should use default disabled', () => {
      const res = Textarea.defaultProps.disabled;
      expect(res).toBe(false);
    });

    it('Should use default name', () => {
      const res = Textarea.defaultProps.name;
      expect(res).toBe('');
    });

    it('Should use default label', () => {
      const res = Textarea.defaultProps.label;
      expect(res).toBe('');
    });

    it('Should use default autoComplete', () => {
      const res = Textarea.defaultProps.autoComplete;
      expect(res).toBe('on');
    });

    it('Should use default autoCorrect', () => {
      const res = Textarea.defaultProps.autoCorrect;
      expect(res).toBe('on');
    });
  });
});
