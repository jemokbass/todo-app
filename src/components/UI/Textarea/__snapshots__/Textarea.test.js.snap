// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea UI component Should render snapshots Should render disabled textarea snapshot 1`] = `
<DocumentFragment>
  <label
    class="label textarea-field"
  >
    <span
      class="textarea-field__label"
    />
    <textarea
      autocomplete="on"
      autocorrect="on"
      class="textarea"
      disabled=""
      name=""
      placeholder=""
    />
  </label>
</DocumentFragment>
`;

exports[`Textarea UI component Should render snapshots Should render empty textarea snapshot 1`] = `
<DocumentFragment>
  <label
    class="label textarea-field"
  >
    <span
      class="textarea-field__label"
    />
    <textarea
      autocomplete="on"
      autocorrect="on"
      class="textarea"
      name=""
      placeholder=""
    />
  </label>
</DocumentFragment>
`;
